tuna/connection.py

```diff
@@ -153,35 +168,43 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       stdout, stderr = self.subp.stdout, self.subp.stderr
       return 0, stdout, stderr
 
+    cmd_idx: int
     for cmd_idx in range(NUM_CMD_RETRIES):
       try:
         if self.ssh is None or not self.ssh.get_transport(
         ) or not self.ssh.get_transport().is_active():
           self.ssh_connect()
+        i_var: Any
+        o_var: Any
+        e_var: None
         i_var, o_var, e_var = self.ssh.exec_command(cmd, timeout=timeout)
+
       except Exception as exc:
-        self.logger.warning('Machine %s failed to execute command: %s', self.id,
-                            cmd)
+        self.logger.warning('Machine %s failed to execute command: %s', id, cmd)
         self.logger.warning('Exception occurred %s', exc)
         self.logger.warning('Retrying ... %s', cmd_idx)
-        retry_interval = randrange(SSH_TIMEOUT)
+        retry_interval = randrange((int)(SSH_TIMEOUT))
         self.logger.warning('sleeping for %s seconds', retry_interval)
         sleep(retry_interval)
       else:
         self.out_channel = o_var.channel
         return i_var, o_var, e_var
 
-      if abort is not None and chk_abort_file(self.id, self.logger):
-        self.logger.warning('Machine %s aborted command execution: %s', self.id,
-                            cmd)
+      if abort is not None and chk_abort_file(id, self.logger):
+        self.logger.warning('Machine %s aborted command execution: %s', id, cmd)
         return None, None, None
 
     self.logger.error('cmd_exec retries exhausted, giving up')
+
     return None, None, None
 
-  def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+
+  def exec_command(self, cmd: str, timeout: int=int(SSH_TIMEOUT), abort:Union[Any, None]=None,\
+  proc_line: Union[Any, None]= None) -> Tuple[Any, int, Any]:
```

Comment:
return values are (integer, StringIO, StringIO)

Reply:
Incompatible return value type (got "Tuple[int, Any, None]", expected "Tuple[int, StringIO, StringIO]")
To solve this error, I had to set it to as above( return type)